functions/Get-DbaAgentAlertCategory.ps1

```diff
@@ -0,0 +1,100 @@
+
+    .NOTES
+        Tags: Agent, Alert, AlertCategory
+        Author: Patrick Flynn (@sqllensman)
+
+        Website: https://dbatools.io
+        Copyright: (c) 2018 by dbatools, licensed under MIT
+        License: MIT https://opensource.org/licenses/MIT
+
+    .LINK
+        https://dbatools.io/Get-DbaAgentAlertCategory
+
+    .EXAMPLE
+        PS C:\> Get-DbaAgentAlertCategory -SqlInstance sql1
+
+        Return all the agent alert categories.
+
+    .EXAMPLE
+        PS C:\> Get-DbaAgentAlertCategory -SqlInstance sql1 -Category 'Severity Alert'
+
+        Return all the agent alert categories that have the name 'Severity Alert'.
+
+    #>
+    [CmdletBinding()]
+    param (
+        [parameter(Mandatory, ValueFromPipeline)]
+        [DbaInstance[]]$SqlInstance,
+        [PSCredential]$SqlCredential,
+        [string[]]$Category,
+        [switch]$EnableException
+    )
+
+    process {
+        foreach ($instance in $SqlInstance) {
+            try {
+                $server = Connect-SqlInstance -SqlInstance $instance -SqlCredential $SqlCredential
+            } catch {
+                Stop-Function -Message "Error occurred while establishing connection to $instance" -Category ConnectionError -ErrorRecord $_ -Target $instance -Continue
+            }
+
+            # get all the alert categories
+            $alertCategories = $server.JobServer.AlertCategories
+            if (Test-Bound -ParameterName Category) {
+                $alertCategories = $alertCategories | Where-Object { $_.Name -in $Category }
+            }
+
+            # Set the default output
+            $defaults = 'ComputerName', 'InstanceName', 'SqlInstance', 'Name', 'ID', 'AlertCount'
+
+            # Loop through each of the categories
+            try {
+                foreach ($cat in $alertCategories) {
+
+                    # Get the alerts associated with the category
+                    $alertCount = ($server.JobServer.Alerts | Where-Object {$_.CategoryName -eq $cat.Name}).Count
+
+                    # Add new properties to the category object
+                    Add-Member -Force -InputObject $cat -MemberType NoteProperty -Name ComputerName -value $server.ComputerName
+                    Add-Member -Force -InputObject $cat -MemberType NoteProperty -Name InstanceName -value $server.ServiceName
+                    Add-Member -Force -InputObject $cat -MemberType NoteProperty -Name SqlInstance -value $server.DomainInstanceName
+                    Add-Member -Force -InputObject $cat -MemberType NoteProperty -Name AlertCount -Value $alertCount
+
+                    # Show the result
+                    Select-DefaultView -InputObject $cat -Property $defaults
+                }
+            } catch {
+                Stop-Function -ErrorRecord $_ -Target $instance -Message "Failure. Collection may have been modified" -Continue
```

Comment:
This seems like a template message...or I just have not seen it used before. Why would we say "may have been modified".